### ISSUE TITLE
JSON field removes empty objects

### ISSUE THREAD

**Issue by chladog** [payloadcms/payload#2218]
# Bug Report

This is reopening of #2189. As after payload@1.6.16 update the empty objects in JSON fields get stripped upon saving the document. I tried clean payload install and the issue is still present.

As the commited fix disabled moongose minimize option I have tested to update the field with local API and **it keeps the empty object correctly**. 

This **works** (```test2: {}``` is present on next load of the document):
```
payload.update({
        id: "64005e1b6811e12c7f73d111",
        collection: 'pages',
        data: {
          test: {
            test1: 1,
            test2: {}
          }
        }
      })
````
Therefore now I suspect it's caused by some processing on Payload's API layer. While sending an update request through REST or GQL it strips the object. (The shape of request's payload coming from admin is correct).

## Steps to Reproduce

1. Have JSON field on Collection
2. Try to store:
```
{
   "test": 1,
   "test2": {}
}
```
4. See the output immediatelly being changed upon saving to:
```
{
   "test": 1
}
```

payload@1.6.16

**Comment by JarrodMFlesch** [comment]
Hey @chladog, I attempted to recreate this issue this morning and was unable to do so. You can see in the screen recording that for me it is saving the empty object properly. 

Can you let me know if anything might be different with your config?

In addition to the recording, I setup a test on a branch with these additional tests: [code comparison](https://github.com/payloadcms/payload/compare/master...test/2218)

https://user-images.githubusercontent.com/30633324/223184024-d43fce61-309f-43ee-81fe-8b7dfcbcdb83.mp4

**Comment by chladog** [comment]
Hello @JarrodMFlesch,
ahha, I tracked the problem...:
So the value gets **stored correctly** in the DB actually and on next refresh it even renders correctly, however if you have **versioning on a collection enabled** and then save the updates on collection item the object is missing in _update response body_ and thats what admin renders resulting in shape described above.

If I remove from collection config following:
```
  versions: {
    drafts: {
      autosave: false,
    },

    maxPerDoc: 50,
  },
```
then it works as in your video.

**Comment by JarrodMFlesch** [comment]
Thanks for the help narrowing it down here @chladog. Fix incoming with #2248 👍
